samples/apply-mosaic-rule-to-rasters/src/main/java/com/esri/arcgismaps/sample/applymosaicruletorasters/components/ApplyMosaicRuleToRastersViewModel.kt

```diff
@@ -0,0 +1,192 @@
+                    Viewpoint(center = imageServiceRasterCenter, scale = 25000.0)
+                )
+                isLoading = false
+            }.onFailure {
+                isLoading = false
+                messageDialogVM.showMessageDialog(it)
+            }
+        }
+    }
+
+    /**
+     * Updates the mosaic rule on the raster layer to the selected [type].
+     */
+    fun updateMosaicRule(type: MosaicRuleType) {
+        _selectedRuleType.value = type
+        isLoading = true
+        // Set the new mosaic rule on the image service raster.
+        imageServiceRaster.mosaicRule = createMosaicRule(type)
+        // Reload the raster layer and update the center point and viewpoint.
+        viewModelScope.launch {
+            rasterLayer.load().onSuccess {
+                mapViewProxy.setViewpointAnimated(
+                    Viewpoint(center = imageServiceRasterCenter, scale = 25000.0)
+                )
+                isLoading = false
+            }.onFailure {
+                isLoading = false
+                messageDialogVM.showMessageDialog(it)
+            }
+        }
+    }
+
+    /**
+     * Returns the center point of the raster service, used for recentering the map.
+     */
+    fun getRasterCenter(): Point = imageServiceRasterCenter
+
+    /**
+     * Helper function to create a [MosaicRule] instance for the given [type].
+     */
+    private fun createMosaicRule(type: MosaicRuleType): MosaicRule {
+        return MosaicRule().apply {
+            when (type) {
+                MosaicRuleType.ObjectID -> {
+                    // Default mosaic method.
+                    mosaicMethod = MosaicMethod.None
+                }
+                MosaicRuleType.NorthWest -> {
+                    // Sorts rasters by northwest location, uses 'first' operation.
+                    mosaicMethod = MosaicMethod.Northwest
+                    mosaicOperation = MosaicOperation.First
+                }
+                MosaicRuleType.Center -> {
+                    // Sorts rasters by proximity to center, uses 'blend' operation.
+                    mosaicMethod = MosaicMethod.Center
+                    mosaicOperation = MosaicOperation.Blend
+                }
+                MosaicRuleType.ByAttribute -> {
+                    // Sorts rasters by the OBJECTID attribute.
+                    mosaicMethod = MosaicMethod.Attribute
+                    sortField = "OBJECTID"
+                }
+                MosaicRuleType.LockRaster -> {
+                    // Locks the mosaic to specific raster IDs.
+                    mosaicMethod = MosaicMethod.LockRaster
+                    lockRasterIds.addAll(listOf(1, 7, 12))
+                }
+            }
+        }
+    }
+
+    companion object {
+        // The sample raster image service URL (Amberg, Germany)
+        private const val IMAGE_SERVICE_URL =
+            "https://sampleserver7.arcgisonline.com/server/rest/services/amberg_germany/ImageServer"
```

Comment:
Can just put this above the class declaration. Doesn't need it's own companion object